modules/margin/src/main/java/com/opengamma/sdk/margin/v3/MarginClient.java

```diff
@@ -76,6 +77,23 @@ public static MarginClient of(ServiceInvoker invoker) {
   public abstract MarginCalcResult calculate(Ccp ccp, MarginCalcRequest request);
 
   /**
+   * High-level call to submit a portfolio for parsing, validation and IM calculation, across multiple CCPs.
+   *
+   * @param ccps the listof CCPs to use for calculation. If one of the selected CCPs is not available to the user, an exception will occur
+   * @param request  the calculation request
+   * @return the detailed result of the calculation, grouped by CCP
+   */
+  public abstract MultiCcpMarginCalcResult calculate(List<Ccp> ccps, MarginCalcRequest request) throws ExecutionException, InterruptedException;
+
+  /**
+   * High-level call to submit a portfolio for parsing, validation and IM calculation, across all available CCPs.
```

Comment:
I'm not convinced that this should be on the API, as it could change its behaviour in a new release when we add a new CCP. Better to force users to choose their list of CCPs.